src/Microsoft.AspNetCore.Blazor/Json/CamelCase.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Blazor.Json
+{
+    internal static class CamelCase
+    {
+        public static string ToCamelCase(this string value)
+        {
+            // Basic cases: if we don't need to modify the value, bail out without creating a char array
+            if (string.IsNullOrEmpty(value))
+            {
+                return value;
+            }
+
+            if (!char.IsUpper(value[0]))
```

Comment:
Are we generally just dealing with things that are valid C# identifiers? or does this need to camel case arbitrary strings?

Reply:
This seems to have the assumption that the input is always already a PascalCase string.

For example, some cases I'm not really sure of the expected output:
`foo_Bar`
`ThisIsXML`

Reply:
OK looks like this is addressed below.

Reply:
Yes, the whole notion of camelcasing is more subjective than first appears. There are plenty of strings where people would disagree about what the expected camelcase verson is, and different libraries do produce different output in practice.

My goal in this case was, for the subset of cases we care about (incoming string is a member identifier), to produce the same result as Json.NET. That's to make it easier for anyone who wants to "grow up" to Json.NET after starting with the built-in utility, or who uses Json.NET on the server and expects their client code to work like other .NET clients.